IHP/IDE/Types.hs

```diff
@@ -147,8 +152,49 @@ data Context = Context
     , portConfig :: PortConfig
     , appStateRef :: IORef AppState
     , isDebugMode :: Bool
+    , frameworkConfig :: FrameworkConfig
+    }
+
+-- Proxies FrameworkConfig fields contained in the RequestContext
+
+configFrameworkConfig :: (?context :: Context) => FrameworkConfig
+configFrameworkConfig = frameworkConfig ?context
+
+configAppHostname :: (?context :: Context) => Text
+configAppHostname = (FrameworkConfig.appHostname . frameworkConfig) ?context
+
+configEnvironment :: (?context :: Context) => Environment
+configEnvironment = (FrameworkConfig.environment . frameworkConfig) ?context
+
+configAppPort :: (?context :: Context) => Int
+configAppPort = (FrameworkConfig.appPort . frameworkConfig) ?context
+
+configBaseUrl :: (?context :: Context) => Text
+configBaseUrl = (FrameworkConfig.baseUrl . frameworkConfig) ?context
+
+configRequestLoggerMiddleware :: (?context :: Context) => Middleware
+configRequestLoggerMiddleware = (FrameworkConfig.requestLoggerMiddleware . frameworkConfig) ?context
+
+configSessionCookie :: (?context :: Context) => Cookie.SetCookie
+configSessionCookie = (FrameworkConfig.sessionCookie . frameworkConfig) ?context
+
+configMailServer :: (?context :: Context) => MailServer
+configMailServer = (FrameworkConfig.mailServer . frameworkConfig) ?context
+
+
+data ContextEqualitySubset = ContextEqualitySubset
+    { actionVar_ :: MVar Action
+    , portConfig_ :: PortConfig
+    , appStateRef_ :: IORef AppState
+    , isDebugMode_ :: Bool
     } deriving (Eq)
 
+toEqualitySubset :: Context -> ContextEqualitySubset
+toEqualitySubset (Context var pconf appstate mode _) = ContextEqualitySubset var pconf appstate mode
+
+instance Eq Context where
```

Comment:
Where's the need for this Eq instance coming from? :) Feels super hacky, but I guess there's a good reason for this.

Reply:
I'm not sure where or why, but equality was previously on Contex. However with the addition of the FrameworkConfig it can't be derived (because Middleware is a function for starters). I'll dig deeper to see if Eq can be omitted.

Reply:
It's because the Context is part of the `ToolServerApplication` of which `frontControllerToWaiApp` requires an Eq instance. However, if I remove that constraint from `frontControllerToWaiApp` everything seems to be fine. Any reason that it is there?